clarity-ext-scripts/clarity_ext_scripts/covid/knm_sminet_service.py

```diff
@@ -0,0 +1,157 @@
+import codecs
+import dateutil.parser
+from datetime import datetime
+from clarity_ext_scripts.covid.partner_api_client import PartnerAPIV7Client
+from sminet_client import SmiNetClient, SmiNetConfig
+from clarity_ext.utils import lazyprop
+from sminet_client import (SampleInfo, ReferringClinic, Patient, Doctor, SmiNetConfig, NotificationType,
+                           LabDiagnosisType, LabResult, Laboratory, SmiNetLabExport)
+from sminet_client import Gender as SmiNetGender
+from clarity_ext_scripts.covid.knm_service import KNMConfig, ServiceRequestProvider
+
+
+class KNMSmiNetIntegrationService(object):
+
+    MAP_GENDER_FROM_KNM_TO_SMINET = {
+        "male": SmiNetGender.MALE,
+        "female": SmiNetGender.FEMALE,
+        "other": SmiNetGender.UNKNOWN,
+        "unknown": SmiNetGender.UNKNOWN,
+        None: SmiNetGender.UNKNOWN,
+    }
+
+    def __init__(self, config):
+        self.config = config
+
+        knm_config = KNMConfig(config)
+        sminet_config = SmiNetConfig(**config)
+
+        self.knm_client = PartnerAPIV7Client(**knm_config)
+        self.sminet_client = SmiNetClient(sminet_config)
+
+    def get_county_from_organization(self, organization):
+        """
+        Given the raw json from the KNM service, returns a county understood by SmiNet
+        """
+        aliases = organization["alias"]
+        for alias in aliases:
+            county_code = alias.split("-")[0]
+            # TODO: THIS IS JUST FOR TEMPORARY TEST PURPOSES (waiting for knm fix)
+            if county_code == "SE":
+                county_code = "AB"
+
+            if self.sminet_client.is_supported_county_code(county_code):
+                return county_code
+        raise NoSupportedCountyCodeFound(
+            "No supported county code found in alias list. Found: {}".format(aliases))
+
+    def create_sample_info(self, provider, sample):
+        sample_date_referral = provider.service_request["resource"]["authoredOn"]
+        # The date is on ISO8601 format:
+        sample_date_referral = dateutil.parser.isoparse(sample_date_referral)
+
+        return SampleInfo(status=1,
+                          sample_id=sample.org_referral_code,
+                          sample_date_arrival=sample.date_arrival,
+                          sample_date_referral=sample_date_referral,
+                          sample_material="Svalg",
+                          sample_free_text_referral="Anamnes: Personalprov")
```

Comment:
I am not sure that it is a good idea to hard code, "Anamnes: Personalprov" here, since I think that the lab runs analysis for other purposes as well.

Reply:
Right, this is currently the only thing we're using and I will drag all the constants up to the extension in the next PR (this one doesn't have the extension bit).